packages/shell-api/src/field-level-encryption.ts

```diff
@@ -0,0 +1,165 @@
+import { classPlatforms, hasAsyncChild, returnsPromise, ShellApiClass, shellApiClassDefault } from './decorators';
+import { Document, ReplPlatform, BinaryType } from '@mongosh/service-provider-core';
+import Collection from './collection';
+import Cursor from './cursor';
+import { DeleteResult } from './result';
+import { assertArgsDefined, assertArgsType } from './helpers';
+import { MongoshInvalidInputError } from '@mongosh/errors';
+
+/** Configuration options for using 'aws' as your KMS provider */
+declare interface awsKms {
+  aws: {
+    /** The access key used for the AWS KMS provider */
+    accessKeyId?: string;
+    /** The secret access key used for the AWS KMS provider */
+    secretAccessKey?: string;
+  };
+}
+
+/** Configuration options for using 'local' as your KMS provider */
+declare interface localKms {
+  local: {
+    /** The master key used to encrypt/decrypt data keys. A 96-byte long Buffer. */
+    key: BinaryType;
+  };
+}
+
+export interface ClientSideFieldLevelEncryptionOptions {
+  keyVaultClient?: any, /** This should be a Mongo class instance, but to avoid circular ref use any **/
+  keyVaultNamespace: string,
+  kmsProvider: awsKms | localKms,
+  schemaMap?: Document,
+  bypassAutoEncryption?: boolean;
+}
+
+@shellApiClassDefault
+@hasAsyncChild
+@classPlatforms([ ReplPlatform.CLI ] )
+export class ClientEncryption extends ShellApiClass {
+  public _mongo: any; // Mongo but any to avoid circular ref
+  public _libmongocrypt: any;
```

Comment:
Is it worth asking the driver team for TS definitions for the libmongocrypt bindings, or something that we would potentially want to do ourselves? (I wouldn’t be opposed to writing them, and the node driver already has some available in its own source tree, but also don’t want to feel like that would be wasting my time)

Reply:
I think I remember asking and they said since it's not part of the driver, they don't write it but I don't think they'd mind us adding them. See this thread for the convo: https://mongodb.slack.com/archives/C0V8RU15L/p1608128060225500. I made this ticket https://jira.mongodb.org/browse/NODE-2989 to see if they'd be alright with us adding the types

Reply:
It seems they're cool with us doing it, so I made a MONGOSH ticket that we can put in the upcoming sprint https://jira.mongodb.org/browse/MONGOSH-527